plugins/q/src/data/extractor-t.js

```diff
@@ -267,7 +267,7 @@ function getHierarchyForSMode(dataset) {
     let isNew = false;
     for (let c = 0; c < dimensions.length; c++) {
       const cell = row[dimensions[c]];
-      const key = `${id}__${cell.qText}`;
+      const key = `${id}__${cell.qText}__${cell.qElemNumber}`;
```

Comment:
maybe
```suggestion
      const key = `${id}__${cell.qElemNumber}`;
```
using qElemNumber, qText should ned be needed.

as the function is called getHierarchyForSMode this is probably only used for cubes when qMode is 'S'
otherwise you need to use {cell.qElemNumber ?? cell.qElemNo}

Reply:
Looks like for mode s, to be safe I changed to` {cell.qElemNumber ?? cell.qElemNo}`